app/components/ProjectRevision/routinelyUpdatedFieldFactory.tsx

```diff
@@ -0,0 +1,147 @@
+import { WidgetProps } from "@rjsf/core";
+import { Button } from "@button-inc/bcgov-theme";
+import React, { useEffect, useMemo, useState } from "react";
+// import { UseDebouncedMutationConfig } from "mutations/useDebouncedMutation"; //TODO: bring this back when we have the mutation
+// import { Disposable, MutationParameters } from "relay-runtime"; //TODO: bring this back when we have the mutation
+import { DateTime } from "luxon";
+
+/**
+ * A factory function that creates a Field/Widget/React component for a routinely updated field.
+ *
+ * @param Widget The widget to wrap like a TextWidget, SelectWidget, etc.
+ * @param mutationFnHook The hook that returns the mutation function and a boolean indicating whether the mutation is in progress
+ * @param fieldNameToUpdate The name of the field to update (e.g. "status")
+ * @param formDataTableName The name of the form data table to use for updating the field (e.g. "project")
+ * @returns a React component that wraps the widget and adds the update button and functionality
+ *
+ * How to use:
+ * 1. Create a mutation function hook that returns the mutation function and a boolean indicating whether the mutation is in progress
+ * 2. Create a field/widget component inside the form component using this factory function
+ * 3. Add the field/widget component to the form widgets prop
+ * 4. Replace the widget in the ui-widget of the UISchema with this newly created field/widget component
+ * 5. Pass the `projectRevision` object to the form context
+ */
+
+const routinelyUpdatedFieldFactory = (
+  Widget: React.FC<WidgetProps>
+  // mutationFnHook: () => [
+  //   (config: UseDebouncedMutationConfig<MutationParameters>) => Disposable,
+  //   boolean
+  // ] //TODO: bring this back when we have the mutation
+  // fieldNameToUpdate: string, //TODO: bring this back when we have the mutation
+  // formDataTableName: string //TODO: bring this back when we have the mutation
+): React.FC<WidgetProps> => {
+  const WrappedWidget: React.FC<WidgetProps> = (props) => {
+    // helper function to compare the initial value to the current value
+    const valueUnchanged = useMemo(
```

Comment:
I'm not sure about the name `valueUnchanged` This is the value that shows up before the user touches anything in the widget? I think `initalValue` like you've got in the comment above might be clearer

Reply:
This is a mini helper function to compare the untouched value with the current value and does some job based on that. I think `initialValue` would conflict with the existing `initialValue` state but any other name would be acceptable :) I'm not good at naming vars :D

Reply:
Oh right, `initialValue` is already taken. Maybe something like... `committedValue`? `previouslyCommittedValue`? But I don't think it's a big deal or a necessary change